src/components/FlightDetails/FlightItineraryInfo.tsx

```diff
@@ -0,0 +1,101 @@
+import React, {useEffect, useState} from "react";
+import {FlightResultsProps} from "../../pages/HomePage/interfaces";
+import {Box, Grid, LinearProgress, Typography} from "@mui/material";
+import {airlineFailoverLogo} from "../FlightCard/FlightInfoComponent";
+import flightCardStyles from "../FlightCard/FlightCard.module.css";
+import {convertDate, timeTravelDiff} from "../../utils/utils";
+import moment from "moment";
+import {getAirlineByCodes, IAirlineInfo} from "../../services/FlightDetails.service";
+
+type FlightItineraryInfoProps = {
+    flightOffer: FlightResultsProps,
+};
+
+const FlightItineraryInfo: React.FC<FlightItineraryInfoProps> = ({flightOffer}: FlightItineraryInfoProps) => {
+    const [airlines, setAirlines] = useState<Map<string, string> | null>(null);
+    const {passengers} = flightOffer;
+    const ItineraryTitle = ["OUTBOUND", "RETURN"];
+
+    const involvedCarrierCodes: string[] = flightOffer.itineraries
+        .map((itinerary) => itinerary.segments[0].carrierCode)
+    const involvedCarrierCodesDistinct = Array.from(new Set(involvedCarrierCodes)).join(',');
+
+    useEffect(()=> {
+        getAirlineByCodes(involvedCarrierCodesDistinct)
+            .then(({data}) =>
+                setAirlines(buildMapOfCodes(data))
+            ).catch((reason) => console.log(reason))
+    }, []);
+
+    const Elements = flightOffer.itineraries.map((itinerary, index) => {
```

Comment:
Can we use a more meaningful variable name like itineraries?, and also we are using camel case so first letter should be lower case.